test/integ_tests/test_create_quantum_job.py

```diff
@@ -227,9 +227,24 @@ def decorator_job(a, b: int, c=0, d: float = 1.0, **extras):
             "extra_arg": "extra_value",
         }
 
-    job = decorator_job(MyClass, 2, d=5, extra_arg="extra_value")
+        with open("output_file.txt", "w") as f:
+            f.write("hello")
+
+    job = decorator_job(MyClass(), 2, d=5, extra_arg="extra_value")
     assert job.result()["status"] == "SUCCESS"
 
+    current_dir = Path.cwd()
+    with tempfile.TemporaryDirectory() as temp_dir:
+        os.chdir(temp_dir)
+        job.download_result()
+        assert (
+            Path(job.name, "results.json").exists()
+            and not Path(job.name, "test", "integ_tests", "requirements.txt").exists()
+        )
+        with open(Path(job.name, "output_file.txt"), "r") as f:
+            assert f.read() == "hello"
+        os.chdir(current_dir)
```

Comment:
What happens if the test fails before this?

Reply:
good point. Copied this pattern from above, so I'll update in both places. The reason for switching the cwd is because download_result downloads to the cwd without an option to specify another location